# Kristin Dowling
# Interactive Data / Spring 2026
## CCI internship data-  Summer 2023 through Fall 2026

In [1]:
import pandas as pd
from matplotlib import pyplot as plt
import plotly.express as px
import dash
import plotly.graph_objects as go 
import pgeocode
import dash
from dash import Dash, html, dcc, Output, Input, callback
#importing everything I will need


*I checked head and it showed the first 5, i just wanted to make sure it was showing the most*
*current internships which it is, I don't remember doing .tail i class but it's what seemed logical*
*to me and it seems like it worked*

In [2]:
intapp = pd.read_csv('Internship Applications_cleanish.csv')
intapp.tail() 

,Submission Date,Advising \nApproval Response,Advising Approval \nResponse Date,Advising Comments,Advisor,Internship Coordinator \nApproval Response,Internship Coordinator \nApproval Response Date,Internship Coordinator Comments,Internship Coordinator,Advising/Tyler Registration Comments,...,Total Work Hours for Semester,Requested Credit Hours,"In-person, Remote, Hybrid",Paid Internship?,Salary,Internship Job Title,Description of Job Responsibilities,How will the internship prepare them for professional practice in their major?,Offered a permanent position (Y/N)?,Accepted permanent position (Y/N)?
942,4/18/26,Approve,2026-04-20T14:17:08Z,NaN,"Hansen, Ellie",Approve,2026-04-20T15:00:18Z,"Please register for 1 credit, summer 2026, BUS...","Knepper, Caitlin",registered for bus 40192-30,...,10.0,1.0,Hybrid,No,NaN,Social Media Marketing Intern,"Assist with creating, scheduling and publishin...",This will help me prepare for what I want to p...,NaN,NaN
943,4/21/26,Approve,2026-04-21T17:06:31Z,NaN,"Hansen, Ellie",Approve,2026-04-21T17:11:54Z,NaN,"Knepper, Caitlin",NaN,...,150.0,1.0,Hybrid,Yes,50$ per article,Reporter,Reporting on issues and events in Portage coun...,"As a journalism major, it is important that I ...",NaN,NaN
944,4/22/26,Approve,2026-04-22T14:17:48Z,NaN,"Hansen, Ellie",Approve,2026-04-22T15:28:22Z,"Confirmed with Tessa, switch to 1 credit","Jeffries, Lauren",NaN,...,150.0,2.0,In-person,No,NaN,Operations and Marketing Intern,"Creating posters/flyers for local bands, makin...",It will be a great way to make connections and...,NaN,NaN
945,4/23/26,N/A - EMAT Graduate Program,N/A - EMAT Graduate Program,N/A - EMAT Graduate Program,N/A - EMAT Graduate Program,Approve,2026-04-24T13:25:14Z,NaN,"Dowling, Kristin",NaN,...,135.0,3.0,In-person,Yes,$15 per hour,Content Producer,planning esports video and content (long form)...,"Everything I will be doing in this internship,...",NaN,NaN
946,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


*How many internship applicants are there?*

In [3]:
len(intapp) 

947

*I wanted to make sure I know all of the columns and need to decide which are most interesting*
*I know the school director wants to see where students are getting internships, I think it would be good to see*
*the numbers of students getting internships by year (but maybe not) I did add the last 2 columns which we do not*
*currently track so these are fake*

In [4]:
intapp.columns 

Index(['Submission Date', 'Advising \nApproval Response',
       'Advising Approval \nResponse Date', 'Advising Comments', 'Advisor',
       'Internship Coordinator \nApproval Response',
       'Internship Coordinator \nApproval Response Date',
       'Internship Coordinator Comments', 'Internship Coordinator',
       'Advising/Tyler Registration Comments', 'Registration Date',
       'Semester Student Registered', 'Student \nFirst Name',
       'Student \nLast Name', 'KSU Email', 'Phone Number', 'CCI Major',
       'Organization Name', 'Description of Organization',
       'Organization Street Address', 'Organization City',
       'Organization State', 'Organization Country (if applicable)',
       'Organization Zip', 'Organization Contact Info', 'Organization Website',
       'Supervisor Name', 'Supervisor Email', 'Supervisor Phone Number',
       'Supervisor Job Title', 'Internship Start Date', 'Internship End Date',
       'Total Number of Weeks Working', 'Work Hours Per Week',
   

*checking what type of data the company names are, this is going to be trouble since it's driven by a form*
*which student input the company name (so there will be multiple variations of the same name)*

In [5]:
type(intapp['Organization Name'].iloc[0])

str

In [6]:
intappdata = pd.DataFrame(intapp['Paid Internship?']) #how do I see if students were paid
intappdata.head(10)

,Paid Internship?
0,No
1,Yes
2,Yes
3,No
4,Yes
5,Yes
6,Yes
7,No
8,No
9,Yes


In [7]:
len(intapp['Paid Internship?'])

947

*I would like to see the majors*

In [8]:
intapp['CCI Major'].unique()

array(['Communication Studies', 'Visual Communication Design (BFA or BA)',
       'Journalism', 'Emerging Media and Technology',
       'Digital Media Production', 'Advertising', 'Public Relations',
       'Photography', 'User Experience Design', nan], dtype=object)

*Now I want to group by major*

In [9]:
paid_unpaid = intapp.groupby('Paid Internship?').size().sort_values()

*I want to make a pie chart showing the paid vs. unpaid internships I played around with the default colors to make it the official KSU specific blue and yellow I initally wanted a pie chart because I thought the statistic was better than 50/50 and would show most students are getting paid internships*

In [10]:
fig1 = go.Figure(
    data=[
        go.Pie(
            labels=paid_unpaid.index,
            values=paid_unpaid.values,
            marker=dict(colors=["#EFAB00", "#003976"])
        )
    ]
)

fig1.update_layout(
    title=dict(
        text='Paid Internship?',
        y=0.95,
        x=0.5,
        xanchor='center'
    ),
    legend=dict(
        x=0.75,
        y=0.5
    ),
    margin=dict(t=80)
)

fig1.show()


*I'm going to use this one for sure, if we had more paid interns it would be good for a*
*"recruiting dashboard" but as is, I think it's a good internal number to have for us to know*

In [11]:
perm_position = intapp.groupby('Offered a permanent position (Y/N)?').size().sort_values()

*I want to show another pie chart with the number of students who were offered permanent posistions after their internship (even though this number is not yet real)*

In [12]:
fig2 = go.Figure(
    data=[
        go.Pie(
            labels=perm_position.index,
            values=perm_position.values,
            title='Offered a permanent position (Y/N)?',
            marker=dict(colors=["#003976"])
 ) ] )
fig2.show()


*I'm going to use this on my final dashboard too but before putting it in the fianl dashboard I need to push the title up a bit, it looks too close to the chart and the legend looks to far away*

*Now I want to take a look at company names so I am going to make a dataframe to take a look at the first ten names*

In [13]:
intappdata = pd.DataFrame(intapp['Organization Name']) 
intappdata.head(10)


,Organization Name
0,KSU / CAPS - Counseling and Psychological serv...
1,Mail Pro 1
2,Stow Munroe Falls City Schools
3,Kent Wired
4,KENT STATE ATHLETIC COMMUNICATIONS
5,KSU/College of Communication and Information
6,Playhouse Square
7,West County News Journal
8,Christ Baptist Church
9,Fox 8


*I want to see how many company names there are*

In [14]:
len(intapp['Organization Name'])

947

*YIKES - 947 companys and 947 applicants. This didn't seem right so I tried  head.(20), (30), and tried 40 but it stopped at 39. I wanted to see duplicates which there were several.*
*I wasn't able to figure out how to clean it here in python so I cleaned it up on the spreadsheet which I know isn't ideal and I would need to find I way to fix this if I plan to implement this. I know this is something the school director would want to see on a "recruiting dashboard"*

*I want to see how many **unique** company names there are now after cleaning it up a little and this is a little better*

In [15]:
intapp['Organization Name'].nunique() 

664

*I just want to glance at some of the names*

In [16]:
intapp['Organization Name'].unique()

array(['KSU / CAPS - Counseling and Psychological services ',
       'Mail Pro 1', 'Stow Munroe Falls City Schools', 'Kent Wired',
       'KENT STATE ATHLETIC COMMUNICATIONS',
       'KSU/College of Communication and Information', 'Playhouse Square',
       'West County News Journal', 'Christ Baptist Church', 'Fox 8',
       'Akron Recording Company', 'Kent State University',
       'Crisis Prevention Institute', 'Ohio Domestic Violence Network',
       'Akron Life Magazine', 'Oriana House',
       'KCD Worldwide Public Relations',
       "SJS Enterprises 426 DBA: Teresa's Pizza",
       'The Pittsburgh Penguins',
       'Kent State Recreation and Wellness Services', 'GRS Recruiting',
       'SpartanNash', "La'Lew Public Relations",
       'Minor Football League (Cleveland Outlaws)',
       'Columbus Christian Center', 'Charles Osowik Management',
       'Todd Biss Productions', 'Summit County Parole Office',
       'Jumpstart Video Productions', '427 Studios',
       'National Multipl

In [17]:
intapp.groupby('Organization Name').count()['Submission Date']
#grouping by company name

Organization Name
 Chambers of Commerce                                         1
 Greater Cleveland Automobile Dealers’ Association (GCADA)    1
 In Step With Horses(ISWH) and American Tire Works(ATW)       1
10k Movement                                                  1
1895 Films                                                    4
                                                             ..
martinwolf                                                    1
ohio employee ownership center                                1
peaches and cream                                             1
red point digital                                             1
s                                                             1
Name: Submission Date, Length: 664, dtype: int64

In [18]:
intapp.groupby('Organization Zip').count()['Submission Date'].sort_values(ascending = False)[0:20]
#grouping by zip code 

Organization Zip
44240    90
44242    87
44304    26
44224    23
44115    22
44313    17
44308    17
44114    17
44139    14
44720    14
44311    13
44243    13
44145    13
44125    12
44266    12
44236    12
44087     9
44107     9
44131     8
44095     7
Name: Submission Date, dtype: int64

*I am going to make a dataframe to look at internships by state*

In [19]:
intappdata = pd.DataFrame(intapp['Organization State'])
intappdata.head(10)
#data frame for state

,Organization State
0,Ohio
1,Ohio
2,Ohio
3,Ohio
4,Ohio
5,Ohio
6,Ohio
7,Pennsylvania
8,Ohio
9,Ohio


*Now I want to group by state so I can drill down to what states most of our students are going to for internships. I'm not surprised to see most are in Ohio which might be good from an economic development standpoint. I am a little surprised to see the next highest is CA and NY. It might bewould be interested to see what majors are going to these states*

In [20]:
intapp.groupby('Organization State').count()['Submission Date'].sort_values(ascending = False)[0:60]
#grouping by state

Organization State
Ohio               739
California          32
New York            31
International       25
Pennsylvania        23
Massachusetts       11
Virginia             9
Wisconsin            7
Colorado             6
Florida              6
Georgia              6
Washington           6
Texas                6
Connecticut          5
Indiana              5
New Jersey           4
Arizona              3
Missouri             3
Michigan             3
Nevada               2
Maryland             2
Kentucky             2
Illinois             2
Washington D.C.      2
North Carolina       1
Maine                1
Oklahoma             1
Louisiana            1
Tennessee            1
Idaho                1
Name: Submission Date, dtype: int64

*Now I want to group by job title so I can start to drill into the what I want on my final dashboard*

In [21]:
intapp.groupby('Internship Job Title').count()['Submission Date'].sort_values(ascending = False)[0:10]
#grouping by job title

Internship Job Title
Graphic Design Intern      44
Intern                     31
Graphic Designer           28
Marketing Intern           25
Social Media Intern        13
Communications Intern      12
Video Production Intern    11
Video Editor               11
Public Relations Intern    11
Photographer               10
Name: Submission Date, dtype: int64

In [22]:
intappdata = pd.DataFrame(intapp['Semester Student Registered'])
intappdata.head(10)
#making a data frame for semester

,Semester Student Registered
0,Fall 2023
1,Summer 2023
2,Fall 2023
3,Fall 2023
4,Fall 2023
5,Fall 2023
6,Summer 2023
7,Fall 2023
8,Fall 2023
9,NaN


In [23]:
intapp.groupby('Semester Student Registered').count()['Submission Date'].sort_values(ascending = False)
#gropu by the number of students per semester

Semester Student Registered
Summer 2024    159
Summer 2025    137
Spring 2026     93
Spring 2024     89
Fall 2025       77
Spring 2025     69
Fall 2024       67
Fall 2023       62
Summer 2026     49
Fall 2026       13
Summer 2023      3
Name: Submission Date, dtype: int64

In [24]:
intapp.groupby('CCI Major').count()['Semester Student Registered'].sort_values(ascending = False)

CCI Major
Visual Communication Design (BFA or BA)    227
Digital Media Production                   192
Journalism                                  99
Public Relations                            78
Communication Studies                       73
Emerging Media and Technology               67
Advertising                                 57
Photography                                 24
User Experience Design                       1
Name: Semester Student Registered, dtype: int64

In [25]:
semester_counts = intapp.groupby('Semester Student Registered').size().sort_index()
#I think Professor Silva helped me with this during office hours

In [26]:
major_counts = intapp.groupby('CCI Major').size().sort_values(ascending=False)

#just going to copy what I did above

*I want to make a chart that shows the number of students doing an internship each semseter by major and I was trying to merge 2 data sets but was ultimately unsuccessful. I hope it's okay to keep this here since I plan to come back to this project*

In [27]:
semester_major_counts = (intapp.groupby(['Semester Student Registered', 'CCI Major']).size().reset_index(name='Count'))
#I want to see the number of students internning by major by semester so I can combine this with the total number of 
# students in each major to ultimately get a % of students interning by semester by major

*Now I would like to create a map of the United States showing where geographically our students are doing their internships*

In [28]:
intapp.groupby('Organization State').count()['Submission Date'].sort_values(ascending = False)[0:60]

Organization State
Ohio               739
California          32
New York            31
International       25
Pennsylvania        23
Massachusetts       11
Virginia             9
Wisconsin            7
Colorado             6
Florida              6
Georgia              6
Washington           6
Texas                6
Connecticut          5
Indiana              5
New Jersey           4
Arizona              3
Missouri             3
Michigan             3
Nevada               2
Maryland             2
Kentucky             2
Illinois             2
Washington D.C.      2
North Carolina       1
Maine                1
Oklahoma             1
Louisiana            1
Tennessee            1
Idaho                1
Name: Submission Date, dtype: int64

In [29]:
state_counts = intapp.groupby('Organization State').size().dropna()
state_counts = state_counts.head(20)
#again copied from above but I only want the top 20

*I learned that it's blank because plotly only recognizes state abreviations for maps, and my spreadsheet has full state names so I pulled the code below from chatgpt to clean it up.*

In [30]:

us_state_abbrev = {
    'Alabama': 'AL','Alaska': 'AK','Arizona': 'AZ','Arkansas': 'AR',
    'California': 'CA','Colorado': 'CO','Connecticut': 'CT','Delaware': 'DE',
    'Florida': 'FL','Georgia': 'GA','Hawaii': 'HI','Idaho': 'ID',
    'Illinois': 'IL','Indiana': 'IN','Iowa': 'IA','Kansas': 'KS',
    'Kentucky': 'KY','Louisiana': 'LA','Maine': 'ME','Maryland': 'MD',
    'Massachusetts': 'MA','Michigan': 'MI','Minnesota': 'MN',
    'Mississippi': 'MS','Missouri': 'MO','Montana': 'MT',
    'Nebraska': 'NE','Nevada': 'NV','New Hampshire': 'NH',
    'New Jersey': 'NJ','New Mexico': 'NM','New York': 'NY',
    'North Carolina': 'NC','North Dakota': 'ND','Ohio': 'OH',
    'Oklahoma': 'OK','Oregon': 'OR','Pennsylvania': 'PA',
    'Rhode Island': 'RI','South Carolina': 'SC','South Dakota': 'SD',
    'Tennessee': 'TN','Texas': 'TX','Utah': 'UT','Vermont': 'VT',
    'Virginia': 'VA','Washington': 'WA','West Virginia': 'WV',
    'Wisconsin': 'WI','Wyoming': 'WY'
}

# clean + convert
intapp['Organization State'] = intapp['Organization State'].str.strip()
intapp['State Abbrev'] = intapp['Organization State'].map(us_state_abbrev)

In [31]:
state_counts = intapp.groupby('State Abbrev').size().dropna()

In [32]:
#tyler helped me get figure out how to play around with this color so it showed the differences bettter 
bluescale = [
    [0.0, "#FFFFFF"],
    [0.05, "#E1CC94"],
    [0.1, "#E0B747"],
    [1, "#EFAB00"]
]

In [33]:
fig7 = go.Figure(data=go.Choropleth(
    locations=state_counts.index,   
    z=state_counts.values,          
    locationmode='USA-states',      
    colorscale= bluescale, #"Blues",
    colorbar_title="Submissions"
))

fig7.update_layout(
    title={
        'text': 'Number of Internships by State Since Fall 2023',
        'x': 0.5,
        'xanchor': 'center'
    },
    geo=dict(
        scope='usa',
        lakecolor='rgb(255, 255, 255)'
    ),
        margin=dict(
        l=10,
        r=10,
        t=40,
        b=10
)
)
fig7.show()
#use this 

*I really like this a lot and played around with it to change the default blue to the offical KSU yellow. I am not shocked to see that most of our internships are in Ohio but was surprised to see that the second most are in CA and NY. I would like to drill down in the future to see what majors are doing these internships*

*Now that I have a nice map of the United States that shows that most of our students are doing their internships in Ohio, I would like to see where specifically in Ohio these are taking place. I think this might be interesting from an economic development standpoint* 

In [34]:
intapp.groupby('Organization Zip').count()['Submission Date'].sort_values(ascending = False)[0:20]

Organization Zip
44240    90
44242    87
44304    26
44224    23
44115    22
44313    17
44308    17
44114    17
44139    14
44720    14
44311    13
44243    13
44145    13
44125    12
44266    12
44236    12
44087     9
44107     9
44131     8
44095     7
Name: Submission Date, dtype: int64

In [35]:
#professor silva reminded me how to do this in class
ohio_data = intapp[intapp['Organization State'] == 'Ohio']

In [36]:
#asked chatgpt how to zoom into ohio further, I had to install pgeocode at my terminal and import it 

nomi = pgeocode.Nominatim('us')

# I guess this makes the zip codes a string
intapp['Organization Zip'] = intapp['Organization Zip'].astype(str).str[:5]

# this gets the lat/lon so glad my professor had mercy on my soul and let me switch to this project
intapp['lat'] = intapp['Organization Zip'].apply(lambda x: nomi.query_postal_code(x).latitude)
intapp['lon'] = intapp['Organization Zip'].apply(lambda x: nomi.query_postal_code(x).longitude)

In [37]:
ohio_data = intapp[intapp['Organization State'] == 'Ohio']
#getting ohio only

In [38]:
zip_counts = ohio_data.groupby(['Organization Zip', 'lat', 'lon']).size().reset_index(name='count')
#grouping by zip codes

In [39]:
#David helped me with this map from https://plotly.com/python/tile-map-layers/

fig8 = px.scatter_map(
    zip_counts,
    lat='lat',
    lon='lon',
    size='count',
    hover_name='Organization Zip'
    #scope='usa'
)
fig8.update_traces(marker=dict(color="#003976"))

fig8.update_layout(
    title='Internship Locations in Ohio (by ZIP Code)',
    geo=dict(
        center={'lat': 40.4173, 'lon': -82.9071},  # center of Ohio
        projection_scale=6  # zoom into Ohio
    )
)
fig8.update_layout(map_style="open-street-map")
fig8.update_layout(margin={"r":0,"t":0,"l":0,"b":0})

fig8.show()

*I really like this plan to use it, I changed the purple to the official KSU blue as suggested by Dr. Silva. I needed to move the title and margins for the final dashboard too*

In [40]:

state_counts_no_oh = state_counts[state_counts.index != 'OH']

In [41]:
state_counts_no_oh

State Abbrev
AZ     3
CA    32
CO     6
CT     5
FL     6
GA     6
ID     1
IL     2
IN     5
KY     2
LA     1
MA    11
MD     2
ME     1
MI     3
MO     3
NC     1
NJ     4
NV     2
NY    31
OK     1
PA    23
TN     1
TX     6
VA     9
WA     6
WI     7
dtype: int64

*Locally hosted dashboard can be found here: http://localhost:8050/*

In [42]:
app = Dash()

app.layout = html.Div([
    dcc.Dropdown(id = 'test'),
    html.H1('Internships in CCI since Fall 2023', style = {'text-align':'center'}),#added title and subtitle
    html.Div([
    # Top left
    html.Div([
        html.Div([
            dcc.Graph(id='fig1') #tried changint this to the title but that didnt' change anything
            ]),
        # Bottom left
        html.Div([
            dcc.Graph(id='fig2')
            ])
        ]),
    # Top right
    html.Div([
        html.Div([
            dcc.Graph(id='fig7')
            ]),
        # Bottom right
        html.Div([
            dcc.Graph(id='fig8')
            ])
        ])
    ], style = {'display': 'flex'})


])
@callback(
     Output('fig1', 'figure'),#tried changing these to titles in various configurations but no change
     Output('fig2', 'figure'),
     Output('fig7', 'figure'),
     Output('fig8', 'figure'),
     Input('test', 'value')
 )
def update_figures(test):
    fig1 = go.Figure(
    data=[
        go.Pie(
            labels=paid_unpaid.index,
            values=paid_unpaid.values,
            marker=dict(colors=["#EFAB00", "#003976"])
        )
    ]
)

    fig1.update_layout(
    title=dict(
        text='Paid Internship?',
        y=0.90,
        x=0.5,
        xanchor='center'
    ),
    legend=dict(
        x=0.75,
        y=1
    ),
    margin=dict(t=80)
)

    fig2 = go.Figure(
    data=[
        go.Pie(
            labels=perm_position.index,
            values=perm_position.values,
            marker=dict(colors=["#003976"])
            )])
    
    fig2.update_layout(
    title=dict(
        text='Offered a permanent position (Y/N)?',
        y=.90,
        x=0.5,
        xanchor='center'
    ),
    legend=dict(
        x=0.75,
        y=1
    ),
    margin=dict(t=80)

    )
    fig7 = go.Figure(data=go.Choropleth(
        locations=state_counts.index,   
        z=state_counts.values,          
        locationmode='USA-states',      
        colorscale= bluescale, #"Blues",
        colorbar_title="Submissions"
        )
    )
    fig7.update_layout(
    title={
        'text': 'Number of Internships by State',
        'x': 0.5,
        'xanchor': 'center'
    },
    geo=dict(
        scope='usa',
        lakecolor='rgb(255, 255, 255)'
    ),
        margin=dict(
        l=10,
        r=10,
        t=40,
        b=10
    )
        )
    fig8 = px.scatter_map(
        zip_counts,
        lat='lat',
        lon='lon',
        size='count',
        hover_name='Organization Zip'
        #scope='usa'
)
    fig8.update_traces(marker=dict(color="#003976"))

    fig8.update_layout(
    title={
        'text': 'Internship Locations in Ohio (by ZIP Code)',
        'x': 0.5,
        'xanchor': 'center',
        'y': 0.95,
        'yanchor': 'top'
    }
    )
    geo=dict(
        center={'lat': 40.4173, 'lon': -82.9071},  # center of Ohio
        projection_scale=6  # zoom into Ohio
    )

    fig8.update_layout(map_style="open-street-map")
    fig8.update_layout(margin={"r":0,"t":50,"l":0,"b":0})
    return fig1, fig2, fig7, fig8

    
if __name__ == '__main__':
    app.run(debug = True)